From 4c53f3472c4140323a16c6f9ba64844a68e2c1fa Mon Sep 17 00:00:00 2001
From: Xinyun Liu <xinyun.liu@intel.com>
Date: Fri, 3 Aug 2018 20:08:50 +0800
Subject: [PATCH 1560/1586] drm/i915/gvt: validate pipe number before use it

SKL_PLANE_REG_TO_PIPE() doesn't ensure pipe be a valid one. So need to
make sure the pipe is in the valid range.

Signed-off-by: Xinyun Liu <xinyun.liu@intel.com>
Reviewed-by: Fei Jiang <fei.jiang@intel.com>
Reviewed-by: Min He <min.he@intel.com>
---
 drivers/gpu/drm/i915/gvt/handlers.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/i915/gvt/handlers.c b/drivers/gpu/drm/i915/gvt/handlers.c
index 8a25568..c54d14e 100644
--- a/drivers/gpu/drm/i915/gvt/handlers.c
+++ b/drivers/gpu/drm/i915/gvt/handlers.c
@@ -868,6 +868,9 @@ static int skl_plane_mmio_write(struct intel_vgpu *vgpu, unsigned int offset,
 	unsigned int pipe = SKL_PLANE_REG_TO_PIPE(offset);
 	unsigned int plane = SKL_PLANE_REG_TO_PLANE(offset);
 
+	if (WARN_ON_ONCE(pipe >= I915_MAX_PIPES))
+		return -EINVAL;
+
 	write_vreg(vgpu, offset, p_data, bytes);
 	if ((vgpu_vreg(vgpu, PIPECONF(pipe)) & I965_PIPECONF_ACTIVE) &&
 			(vgpu->gvt->pipe_info[pipe].plane_owner[plane] == vgpu->id)) {
-- 
2.7.4

